### PR TITLE
Improved ci

### PR DESCRIPTION
**Pull request by AdrieanKhisbe**
Improved continous integration with travis container and coverage.

Update dependencies and add New badges to README.

:gift:

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,51 @@
...
+      - g++-4.8
+      - libcairo2-dev
+      - libjpeg8-dev
+      - libpango1.0-dev
+      - libgif-dev
+      - build-essential
+
+before_install:
+  - npm -g install npm@latest-2
+  - npm explore npm -g -- npm install node-gyp@latest
```
Comment (espadrine) [inline .travis.yml]: Out of curiosity, could you detail what that does?

File: package.json [hunk 2]
```diff
@@ -48,5 +58,7 @@
     "lru-cache.js",
     "logo"
   ],
-  "engines": { "node": "0.10.x" }
+  "engines": {
+    "node": ">=0.10.0"
```
Comment (espadrine) [inline package.json]: Are you sure node v>=0.10.0 is supported?

File: package.json [hunk 3]
```diff
@@ -17,26 +22,31 @@
   },
   "dependencies": {
     "dot": "~1.0.3",
-    "svgo": "~0.5.1",
-    "canvas": "~1.1.2",
+    "svgo": "~0.6.1",
+    "canvas": "~1.3.5",
```
Comment (espadrine) [inline package.json]: That breaks upon `npm install` on my machine (Linux node v5.4.1 npm v3.3.12) with some nan error `‘class v8::String’ has no member named ‘GetExternalAsciiStringResource’`.

Historically, upgrading canvas has been a phenonculously difficult task. I recommend skipping that.

File: package.json [hunk 4]
```diff
@@ -17,26 +22,31 @@
...
   },
   "scripts": {
-    "test": "mocha -R spec test.js"
+    "test": "mocha -R spec test.js",
+    "coverage": "istanbul cover _mocha test.js; cat ./coverage/lcov.info | coveralls"
+  },
+  "bin": {
+    "badge": "./gh-badge.js"
   },
-  "bin": { "badge": "./gh-badge.js" },
```
Comment (espadrine) [inline package.json]: Why do you remove that?

File: package.json [hunk 5]
```diff
@@ -17,26 +22,31 @@
...
-    "mocha": "~1.14.0"
+    "coveralls": "^2.11.6",
+    "istanbul": "^0.4.1",
+    "mocha": "~2.3.4",
+    "should": "~8.0.2"
   },
   "scripts": {
-    "test": "mocha -R spec test.js"
+    "test": "mocha -R spec test.js",
+    "coverage": "istanbul cover _mocha test.js; cat ./coverage/lcov.info | coveralls"
```
Comment (espadrine) [inline package.json]: Coverage is already dealt with as part of running the test. Adding istanbul is not necessary.

File: package.json [hunk 6]
```diff
@@ -17,26 +22,31 @@
...
+    "semver": "~5.1.0",
+    "bower": "~1.7.1",
+    "promise": "~7.1.1"
   },
   "devDependencies": {
     "ass": "~0.0.6",
-    "should": "~3.0.0",
-    "mocha": "~1.14.0"
+    "coveralls": "^2.11.6",
+    "istanbul": "^0.4.1",
```
Comment (espadrine) [inline package.json]: As mentioned, coverage is already part of the test. coveralls and istanbul are not necessary.

**Comment by espadrine** [comment]
Thanks for contributing!

Could you squash the commits into a single one, as the whole change is fairly small?

File: .travis.yml (hunk 1 above)
Comment (AdrieanKhisbe) [inline .travis.yml]: Travis new infrastructure is running on a container infrastructure.
Synce we need a C++ compiler, to build native module, we need to install it. (with the needed libraries for canvas)

File: package.json (hunk 2 above)
Comment (AdrieanKhisbe) [inline package.json]: you mean the 0.0 patch version?

File: package.json (hunk 3 above)
Comment (AdrieanKhisbe) [inline package.json]: okey dokey, will change that.

File: package.json (hunk 6 above)
Comment (AdrieanKhisbe) [inline package.json]: I don't see what you mean

**Comment by AdrieanKhisbe** [comment]
about squashing, I don't recommend it. -> that helps identifying falling part of the CI build, and attach change to reason. (for future ci modification if needed)
But it's your call (confirm or not along with the other questions), and I'll do. :smiley:

File: package.json (hunk 2 above)
Comment (espadrine) [inline package.json]: I mean, for instance, node v5.4.1, which doesn't seem to give acceptable results at least on my machine. The badge width computation is wrong.

File: package.json (hunk 6 above)
Comment (espadrine) [inline package.json]: I mean that npm test computes coverage information. See the end of test.js.

**Comment by espadrine** [comment]
> about squashing, I don't recommend it

What I need commits to be are to help me when I inevitably have to do some archaeological exploration or bisect. Having a single PR (which should correspond to solving a single issue) include multiple commits involved in solving the same piece of the puzzle makes it difficult to see how they relate. Typically, here, https://github.com/AdrieanKhisbe/shields/commit/c8e083b7f51d4c36d712f8148a4b4bf5b54e5be1 and https://github.com/AdrieanKhisbe/shields/commit/d3e06b3e0191c74ef0f2643ef703861ae3297190. I could go through the work of explaining exactly which commits should be squashed together, but it would take a long time. For a PR this simple, a single commit is really not an issue.

> that helps identifying falling part of the CI build

Every commit should make the build pass. Otherwise, git bisect won't work.

**Comment by paulmelnikow** [comment]
Thanks to this, and the context from the discussion, most of these changes have been merged. Containers were added in #768. PRs are flowing, so please free to open a new PR if there are additional bits worth adding. Closing.

**Pull request closed without merging**